packages/react/src/components/checkbox.tsx

```diff
@@ -0,0 +1,122 @@
+    "focus-visible:ring-[3px] focus-visible:ring-fr-primary",
+    "focus:ring-[3px] focus:ring-fr-primary",
+    "disabled:opacity-50 disabled:cursor-not-allowed disabled:bg-bg-washed data-[state=checked]:disabled:bg-primary-600",
+  ],
+  {
+    variants: {
+      size: {
+        small: "size-4",
+        medium: "size-5",
+      },
+    },
+    defaultVariants: {
+      size: "small",
+    },
+  },
+);
+
+interface CheckboxProps
+  extends React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>,
+    VariantProps<typeof checkboxVariants> {}
+
+export const Checkbox = React.forwardRef<
+  React.ElementRef<typeof CheckboxPrimitive.Root>,
+  CheckboxProps
+>(({ className, size, ...props }, ref) => {
+  return (
+    <CheckboxPrimitive.Root
+      ref={ref}
+      className={clx(checkboxVariants({ size, className }))}
+      {...props}
+    >
+      <CheckboxPrimitive.Indicator
+        className={clx(
+          "flex items-center justify-center text-white",
+          size === "medium" ? "size-4" : "size-3",
+          "[&_.minus-icon]:hidden",
+          "[&_.checked-icon]:hidden",
+          "[&[data-state=checked]_.checked-icon]:inline-block",
+          "[&[data-state=indeterminate]_.minus-icon]:inline-block",
+        )}
+      >
+        <MinusIcon
+          className={clx(
+            size === "medium" ? "stroke-[2.0px]" : "stroke-[1.5px]",
+            "minus-icon",
+          )}
+        />
+        <CheckIcon
+          className={clx(
+            size === "medium" ? "stroke-[2.0px]" : "stroke-[1.5px]",
+            "checked-icon",
+          )}
+        />
+      </CheckboxPrimitive.Indicator>
+    </CheckboxPrimitive.Root>
+  );
+});
+
+Checkbox.displayName = "Checkbox";
+
+const CheckboxContext = React.createContext<{
+  size?: "small" | "medium" | "large";
+  disabled?: boolean;
+}>({
+  size: "small",
+  disabled: false,
+});
+
+export const CheckboxItem: React.FC<
+  React.PropsWithChildren<{
+    size?: "small" | "medium" | "large";
+    disabled?: boolean;
+  }>
+> = ({ children, size = "small", disabled = false }) => {
+  return (
+    <CheckboxContext.Provider value={{ size, disabled }}>
+      {children}
+    </CheckboxContext.Provider>
+  );
+};
+
+export const CheckboxLabel = ({
+  className,
+  children,
+  ...props
+}: React.ComponentProps<"label">) => {
+  const { size, disabled } = React.useContext(CheckboxContext);
+
+  return (
+    <label
+      className={clx(
+        "text-txt-black-700 px-2 font-medium",
+        size === "small" && "text-body-sm",
+        size === "medium" && "text-body-md",
+        size === "large" && "text-lg",
+        disabled && "text-txt-black-disabled",
+        className,
+      )}
+      {...props}
+    >
+      {children}
+    </label>
+  );
+};
+
+CheckboxLabel.displayName = "CheckboxLabel";
```

Comment:
I am looking to remove this. Including the label in `toggle.tsx`. Label will be a standalone component

Reply:
Roger boss. Noted.